src/keymap/keymap.js

```diff
@@ -0,0 +1,184 @@
+     * elements on the page (file list, comments, patches).
+     * This provides compatibility with BitBucket's built in key binds
+     * so that we don't break the existing functionality of the 'j' and 'k' keys.
+     *
+     */
+    self.focusComment = function(comment) {
+        $(self.iterableItemSelector).removeClass('focused');
+        $(comment).addClass('focused');
+        comment.scrollIntoView();
+    };
+
+    /**
+     * Initializes array of comments elements to cycle through using
+     * {@link #scrollToNextComment} or {@link #scrollToPreviousComment}.
+     *
+     * Waits for `selector` to be available before initializing comments.
+     *
+     * @param {String} selector waits for this selecor to become available
+     * before initializing comments. If empty, uses the default `.bb-patch`
+     * selector.
+     */
+    self.initComments = function(selector = '.bb-patch') {
+        waitForRender(selector).then(() => {
+            self.comments = document.querySelectorAll(self.commentSelector);
+        });
+    };
+
+    /**
+     * Scrolls the browser window to the previous comment on the PR diff.
+     *
+     * If scrolling to the next comment from the last comment, this will loop back
+     * to the top-most (first) comment on the page.
+     *
+     */
+    self.scrollToNextComment = function() {
+        if (self.comments) {
+            $(self.comments[self.currentComment]).removeClass('focused');
+            self.currentComment++;
+            if (self.currentComment >= self.comments.length) self.currentComment = 0;
+
+            var comment = self.comments[self.currentComment].parentElement.parentElement;
+            self.focusComment(comment);
+        }
+    };
+
+    /**
+     * Scrolls the browser window to the previous comment on the PR diff.
+     *
+     * If scrolling to previous comment from the top commit, this will loop back to the
+     * bottom-most (last) comment on the page.
+     */
+    self.scrollToPreviousComment = function() {
+        if (self.comments) {
+            self.currentComment--;
+            if (self.currentComment < 0) self.currentComment = self.comments.length - 1;
+
+            var comment = self.comments[self.currentComment].parentElement.parentElement;
+            self.focusComment(comment);
+        }
+    };
+
+    /**
+     * Initializes keybinds.
+     *
+     * @param {Object} keyboard the keyboard library to use to bind keys (usually Mousetrap).
+     * @param {Object} user_keymap the user-defined keymap to override default keybindings.
+     */
+    self.init = function(keyboard, userKeymap) {
+
+        Object.assign(keymap, default_keymap);
+        if (userKeymap) {
+            //if provided, copy a user-preferred keymap to the main keymap.
+            Object.assign(keymap, userKeymap);
+        }
+
+        keyboard.reset();
+        self.comments = document.querySelectorAll(self.commentSelector);
```

Comment:
You're right, that one slipped out unnoticed. I moved it around a bunch trying to get it to work in the original code and this must've been left behind.

I'm not sure about the waitForRender... Would the waitForRender then trigger the entire keymap? The only keys we need to wait for rending on are the comment keys, the tab changing and scroll to top and bottom should still work. The selector we are waiting for also changes depending on the tab we have activated and will need to wait again when we switch back and forth between *Overview* and *Activity* tabs.